### PR TITLE
Add support for reading LZ4 and ZSTD-compresed parquet data

### PR DESCRIPTION
**Pull request by shangxinli**
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by v-jizhang** [comment]
@shangxinli Should we specify `Lz4Codec.class `and `ZStandardCodec.class`? They are `nulls `in HiveCompressionCodec.java
```
LZ4(null, CompressionKind.NONE, null, f -> f == PAGEFILE),
ZSTD(null, CompressionKind.ZSTD, null, f -> f == ORC || f == DWRF || f == PAGEFILE);
```

**Comment by v-jizhang** [comment]
@shangxinli This is a backport from https://github.com/trinodb/trino/pull/910, please add co-author. https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors

**Comment by shangxinli** [comment]
> @shangxinli Should we specify `Lz4Codec.class `and `ZStandardCodec.class`? They are `nulls `in HiveCompressionCodec.java
> 
> ```
> LZ4(null, CompressionKind.NONE, null, f -> f == PAGEFILE),
> ZSTD(null, CompressionKind.ZSTD, null, f -> f == ORC || f == DWRF || f == PAGEFILE);
> ```

@v-jizhang This seems unrelated. Correct me if I am wrong. ZStandardCodec.class is for Luben ZSTD and this PR is using airlift ZSTD.

**Comment by shangxinli** [comment]
Sorry @v-jizhang, I overlooked the namespace of ZStandardCodec. It is from Hadoop, not from Parquet(ZStandardCodec in Parquet was added later). 

There were 3 commits in original PR. We should port the two(parquet and orc ZSTD). For the 3rd one, because prestodb is still using lower version of Hadoop, which doesn't have ZStandardCodec yet.

**Comment by v-jizhang** [comment]
@shangxinli Please fix the test errors. Thanks

**Comment by v-jizhang** [comment]
Is it possible to test the new features? otherwise LGTM. Thanks

**Comment by shangxinli** [comment]
We have deployed it in production for ~1 year.

**Comment by v-jizhang** [comment]
> 
> 
> We have deployed it in production for ~1 year.

I meant to add a unit test. Thanks

**Comment by beinan** [comment]
> Sorry @v-jizhang, I overlooked the namespace of ZStandardCodec. It is from Hadoop, not from Parquet(ZStandardCodec in Parquet was added later).
> 
> There were 3 commits in original PR. We should port the two(parquet and orc ZSTD). For the 3rd one, because prestodb is still using lower version of Hadoop, which doesn't have ZStandardCodec yet.
> 
> Added co-author @martint.

Twitter has upgraded presto-hadoop-apache to use hadoop-common 2.9 (with a native libzstd loaded), it's running on prod now.  I might send a PR later if everything goes well.  Then we can port the 3rd one I guess.

File: presto-parquet/src/test/java/com/facebook/presto/parquet/TestParquetCompressionUtils.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        performTest(CompressionCodecName.ZSTD, 512);
+        performTest(CompressionCodecName.ZSTD, 1024);
+        performTest(CompressionCodecName.ZSTD, 2048);
+    }
+
+    @Test
+    public void testDecompressLZ4()
+            throws IOException
+    {
+        performTest(CompressionCodecName.LZ4, 0);
```
Comment (beinan) [inline presto-parquet/src/test/java/com/facebook/presto/parquet/TestParquetCompressionUtils.java:47]: static import CompressionCodecName.LZ4

**Review by beinan — APPROVED** [review]

**Comment by beinan** [comment]
> @beinan Can you help to kick off the tests? I guess first time contributor is always blocked.

could you rebase the master first?  looks like there is a conflict.  thanks!

**Comment by beinan** [comment]
> > @beinan Can you help to kick off the tests? I guess first time contributor is always blocked.
> 
> could you rebase the master first? looks like there is a conflict. thanks!

@shangxinli could you help to rebase master and resolve the conflict first?  Thanks!

**Pull request closed without merging**